recipes/imake/all/conanfile.py

```diff
@@ -109,32 +100,41 @@ def _configure_autotools(self):
             "--enable-revpath={}".format(yes_no(self.options.revpath)),
             "--enable-xmkmf={}".format(yes_no(self.options.xmkmf)),
         ]
-
-        # FIXME: RAWCPP (ac_cv_path_RAWCPP) is not compatible with MSVC preprocessor. It needs to be cpp.
-        if tools.get_env("CPP"):
+        if "CPP" in os.environ:
             conf_args.extend([
-                "--with-script-preproc-cmd={}".format(tools.get_env("CPP")),
+                "--with-script-preproc-cmd={}".format(os.environ["CPP"]),
             ])
-        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-        return self._autotools
+
+        env = tc.environment()
+        if is_msvc(self):
+            compile_wrapper = unix_path(self, self.conf.get('user.automake:compile-wrapper'))
+            env.define("CC", f"{compile_wrapper} cl -nologo")
+            env.define("CXX", f"{compile_wrapper} cl -nologo")
+            env.define("CPP", f"{compile_wrapper} cl -E")
+            # We may be able to use AutotoolsDeps, however there are outstanding
+            # issues with path conversions: https://github.com/conan-io/conan/issues/12784
+            xorg_proto_include = unix_path(self, self.dependencies['xorg-proto'].cpp_info.aggregated_components().includedirs[0])
+            env.append("CFLAGS", f"-I{xorg_proto_include}")
+        tc.generate(env)
+
+        pkgconf = PkgConfigDeps(self)
+        pkgconf.generate()
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        # tools.replace_in_file(os.path.join(self._source_subfolder, ""))
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.make(args=["V=1"])
+        apply_conandata_patches(self)
+        autotools = Autotools(self)
+        autotools.configure()
+        autotools.make(args=["V=1"])
 
     def package(self):
-        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.install()
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+        copy(self, "COPYING", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        autotools = Autotools(self)
+        autotools.install()
```

Comment:
Is this working in 1.53 or does it need the 1.54 fix for the "destdir" 🤔 I know there was something here in the past